src/file.cr

```diff
@@ -355,6 +364,11 @@ class File < IO::FileDescriptor
     (stat.st_mode & LibC::S_IFMT) == LibC::S_IFLNK
   end
 
+   # Opens the file named by filename. If a file is being created, its initial
+   # permissions may be set using the perm parameter.
+   # If the optional code block is given, it will be passed the opened file as
```

Comment:
Almost there :-)

Here the block is not optional, it's the overload that accepts a block. So the docs should probably say something like "opens the file..., yields it to the block, ..., closes the file at the end of the block"